### PR TITLE
[TOOLS-1960] Add bugdet information to the project - BE

### PR DESCRIPTION
**Pull request by emannuelalmeida**

### CODE REVIEW

File: src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import objective.taskboard.domain.ProjectFilterConfiguration;
+import objective.taskboard.repository.ProjectFilterConfigurationCachedRepository;
+
+@Service
+public class ChangeRequestService {
+
+    private static final Logger log = LoggerFactory.getLogger(ChangeRequestService.class);
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java]: MAJOR Remove this unused "log" private field. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1068)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestJpaRepository.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import objective.taskboard.domain.ProjectFilterConfiguration;
+
+@Repository
+@Transactional(readOnly = true)
+public class ChangeRequestJpaRepository implements ChangeRequestRepository {
+
```
Comment (jhonypg) [inline src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestJpaRepository.java]: Remover um dos espaços.

File: src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import objective.taskboard.domain.ProjectFilterConfiguration;
+import objective.taskboard.repository.ProjectFilterConfigurationCachedRepository;
+
```
Comment (jhonypg) [inline src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java]: Remover os imports desnecessários.

File: src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    @Autowired
+    public ChangeRequestService(ChangeRequestRepository changeRequestRepository) {
+        this.changeRequestRepository = changeRequestRepository;
+    }
+
+    public List<ChangeRequest> listByProject(ProjectFilterConfiguration project) {
+
+        List<ChangeRequest> items = changeRequestRepository.listByProject(project);
+        return items;
+    }
```
Comment (jhonypg) [inline src/main/java/objective/taskboard/project/config/changeRequest/ChangeRequestService.java]: Adicionar os métodos necessários para ao menos atendermos o "the service must not allow removal of the baseline change order". E criar um teste para ele.

File: src/test/java/objective/taskboard/project/config/changeRequest/ChangeRequestControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+import objective.taskboard.testUtils.ControllerTestUtils.AssertResponse;
+
+public class ChangeRequestControllerTest {
+
+    @Test
+    public void getAll_ifProjectExists_returnOkWithAllProjectsChangeRequests() {
+        ChangeRequestController subject = new ChangeRequestControllerBuilder()
+                .withAuthorizedProject("TASKB")
+                .withChangeRequest("baseline", parseStringToDate("2018-12-28"), 250)
+                .withChangeRequest("Change Request 1", parseStringToDate("2018-12-30"), 30)
```
Comment (jhonypg) [inline src/test/java/objective/taskboard/project/config/changeRequest/ChangeRequestControllerTest.java]: Coloca o "parseStringToDate" dentro do método "withChangeRequest"

File: src/test/java/objective/taskboard/project/config/changeRequest/ChangeRequestControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+                .withNonAuthorizedProject("TASKB")
+                .build();
+        
+        AssertResponse.of(subject.getAll("TASKB"))
+            .httpStatus(NOT_FOUND)
+            .emptyBody();
+        
+    }
+
+    private static class ChangeRequestControllerBuilder {
```
Comment (jhonypg) [inline src/test/java/objective/taskboard/project/config/changeRequest/ChangeRequestControllerTest.java]: Realizar as mudanças conversadas.

File: src/main/java/objective/taskboard/utils/DateTimeUtils.java [hunk 7]
```diff
@@ -145,6 +145,11 @@ public static Date parseStringToDate(String yyyymmdd) {
         }
         return null;
     }
+    
+    
+    public static long parseDateStringAsMilliseconds(String date) {
+        return parseStringToDate(date).getTime();
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/utils/DateTimeUtils.java]: MAJOR A "NullPointerException" could be thrown; "parseStringToDate" is nullable here. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2259)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/main/resources/db/migration/mysql/V201811131033__createChangeRequestTable.sql [hunk 8]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE change_request (
+    id bigint(20) NOT NULL AUTO_INCREMENT,
+    project_id bigint(20) NOT NULL,
+    `date` datetime NOT NULL,
+    name VARCHAR(200) NOT NULL,
+    budget_increase int NOT NULL,
+    isBaseline BOOLEAN NOT NULL,
```
Comment (jhonypg) [inline src/main/resources/db/migration/mysql/V201811131033__createChangeRequestTable.sql]: is_baseline

File: src/main/resources/db/migration/oracle/V201811131033__createChangeRequestTable.sql [hunk 9]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE change_request (
+    id NUMBER NOT NULL,
+    project_id NUMBER NOT NULL,
+    "DATE" date NOT NULL,
+    name varchar(255 char) NOT NULL,
+    budget_increase NUMBER(10,0) NOT NULL,
+    isBaseline NUMBER(1) NOT NULL,
```
Comment (jhonypg) [inline src/main/resources/db/migration/oracle/V201811131033__createChangeRequestTable.sql]: is_baseline

File: src/test/java/objective/taskboard/project/config/changeRequest/ChangeRequestServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import objective.taskboard.domain.ProjectFilterConfiguration;
+
+public class ChangeRequestServiceTest {
+
+    @Rule
+    public ExpectedException thrown= ExpectedException.none();
+    private static ProjectFilterConfiguration project = mock(ProjectFilterConfiguration.class);
```
Comment (jhonypg) [inline src/test/java/objective/taskboard/project/config/changeRequest/ChangeRequestServiceTest.java]: Remover esse "static"

**Pull request merged**